# Statistika Deskriptif - Standar Deviasi dan Variasi

Identitas diri : Ruth Septriana Sipangkar_24060124120024 (E)

Untuk mempelajari lebih lanjut mengenai properti dari standar deviasi dan variasi, kita gunakan library Panda dan dataset berikut.

In [ ]:
import pandas as pd

# Unduh file CSV jika belum ada di direktori saat ini
!pip install gdown
!gdown --id 1KEwyokOzuHi28Td8vZETxstNlmMRE8dC

# Membaca file CSV yang ada di direktori sebagai DataFrame
df = pd.read_csv("Latihan.csv")

Dataset "Latihan" merupakan dataset yang berisi sampel jumlah jawaban benar untuk masing-masing tipe ujian dan skornya pada suatu latihan suatu mata kuliah tertentu.

## Properti Asli

Mari kita hitung nilai standar deviasi dan variasi dari data frame "Latihan", untuk kali ini saya akan menggunakan dataset **Tipe 1**.

In [2]:
# Menampilkan 5 baris pertama dari DataFrame
print(df.head())

# Menghitung standar deviasi dan variasi untuk dataset "Tipe 1"
std_tipe1 = df['Tipe1'].std()
var_tipe1 = df['Tipe1'].var()

# Menampilkan hasil standar deviasi dan variasi
print("Standar deviasi dari dataset Tipe 1:", std_tipe1)
print("Variasi dari dataset Tipe 1:", var_tipe1)

   Tipe1  Tipe2  Tipe3  Tipe4
0      5      5      4      2
1      5      3      2      3
2      5      4      2      3
3      5      3      2      2
4      4      3      2      4
Standar deviasi dari dataset Tipe 1: 4.413931604340082
Variasi dari dataset Tipe 1: 19.482792207792205


## Properti 1: Menambah atau Mengurangi semua Dataset dengan Nilai Konstanta

### Langkah 1: Memodifikasi dataset

Misal kita lakukan modifikasi kepada setiap data di dataset dengan **penjumlahan 2** sebagai modifikasi1 dan **pengurangan 3** sebagai modifikasi2.

In [3]:
# Menambahkan 2 ke setiap elemen sebagai modifikasi1
df_modif1 = df+2
# Menampilkan 5 baris pertama dari dataset modifikasi1
print("Dataset modifikasi1:")
print(df_modif1.head())

# Mengurangkan 3 ke setiap elemen sebagai modifikasi2
df_modif2 = df-3
# Menampilkan 5 baris pertama dari dataset modifikasi2
print("\nDataset modifikasi2:")
print(df_modif2.head())

Dataset modifikasi1:
   Tipe1  Tipe2  Tipe3  Tipe4
0      7      7      6      4
1      7      5      4      5
2      7      6      4      5
3      7      5      4      4
4      6      5      4      6

Dataset modifikasi2:
   Tipe1  Tipe2  Tipe3  Tipe4
0      2      2      1     -1
1      2      0     -1      0
2      2      1     -1      0
3      2      0     -1     -1
4      1      0     -1      1


### Langkah 2: Menghitung nilai standar deviasi dan variasi dari data yang sudah dimodifikasi

In [4]:
# Menghitung standar deviasi dan variasi untuk modifikasi1 di dataset "Tipe 1"
std_modif1 = df_modif1['Tipe1'].std()
var_modif1 = df_modif1['Tipe1'].var()
# Menampilkan hasil standar deviasi dan variasi
print("Standar deviasi dari modifikasi 1 dataset Tipe 1:", std_modif1)
print("Variasi dari modifikasi 1 dataset Tipe 1:", var_modif1)

# Menghitung standar deviasi dan variasi untuk modifikasi2 di dataset "Tipe 1"
std_modif2 = df_modif2['Tipe1'].std()
var_modif2 = df_modif2['Tipe1'].var()
# Menampilkan hasil standar deviasi dan variasi
print("\nStandar deviasi dari modifikasi 2 dataset Tipe 1:", std_modif2)
print("Variasi dari modifikasi 2 dataset Tipe 1:", var_modif2)

Standar deviasi dari modifikasi 1 dataset Tipe 1: 4.413931604340082
Variasi dari modifikasi 1 dataset Tipe 1: 19.48279220779221

Standar deviasi dari modifikasi 2 dataset Tipe 1: 4.413931604340082
Variasi dari modifikasi 2 dataset Tipe 1: 19.482792207792205


### Langkah 3: Analisis

Dari beberapa perhitungan diatas kita peroleh hasil sebagai berikut:

Properti asli                : standar deviasi 4.41 dan variasi 19.48

Modifikasi 1 (penjumlahan 2) : standar deviasi 4.41 dan variasi 19.48

Modifikasi 2 (pengurangan 3) : standar deviasi 4.41 dan variasi 19.48

Dapat kita lihat meskipun sudah dimodifikasi dengan penjumlahan dan pengurangan pada setiap data dalam dataset, standar deviasi dan variasi tidak berubah. Itu artinya perubahan nilai ini tidak akan mempengaruhi distribusi data. 

Sehingga bisa kita simpulkan bahwa penjumlahan dan pengurangan dataset tidak memengaruhi standar deviasi dan variasi dataset. 

## Properti 2: Mengalikan atau Membagi semua Dataset dengan Nilai Konstanta

### Langkah 1: Memodifikasi dataset

Misal kita lakukan modifikasi kepada setiap data di dataset dengan **perkalian 2** sebagai modifikasi1 dan **pembagian 3** sebagai modifikasi2.

In [5]:
# Menglikan 2 ke setiap elemen sebagai modifikasi1
df_modif1 = df*2
# Menampilkan 5 baris pertama dari dataset modifikasi1
print("Dataset modifikasi1:")
print(df_modif1.head())

# Membagi 3 ke setiap elemen sebagai modifikasi2
df_modif2 = df/3
# Menampilkan 5 baris pertama dari dataset modifikasi2
print("\nDataset modifikasi2:")
print(df_modif2.head())

Dataset modifikasi1:
   Tipe1  Tipe2  Tipe3  Tipe4
0     10     10      8      4
1     10      6      4      6
2     10      8      4      6
3     10      6      4      4
4      8      6      4      8

Dataset modifikasi2:
      Tipe1     Tipe2     Tipe3     Tipe4
0  1.666667  1.666667  1.333333  0.666667
1  1.666667  1.000000  0.666667  1.000000
2  1.666667  1.333333  0.666667  1.000000
3  1.666667  1.000000  0.666667  0.666667
4  1.333333  1.000000  0.666667  1.333333


### Langkah 2: Menghitung nilai standar deviasi dan variasi dari data yang sudah dimodifikasi

In [6]:
# Menghitung standar deviasi dan variasi untuk modifikasi1 di dataset "Tipe 1"
std_modif1 = df_modif1['Tipe1'].std()
var_modif1 = df_modif1['Tipe1'].var()
# Menampilkan hasil standar deviasi dan variasi
print("Standar deviasi dari modifikasi 1 dataset Tipe 1:", std_modif1)
print("Variasi dari modifikasi 1 dataset Tipe 1:", var_modif1)

# Menghitung standar deviasi dan variasi untuk modifikasi2 di dataset "Tipe 1"
std_modif2 = df_modif2['Tipe1'].std()
var_modif2 = df_modif2['Tipe1'].var()
# Menampilkan hasil standar deviasi dan variasi
print("\nStandar deviasi dari modifikasi 2 dataset Tipe 1:", std_modif2)
print("Variasi dari modifikasi 2 dataset Tipe 1:", var_modif2)



Standar deviasi dari modifikasi 1 dataset Tipe 1: 8.827863208680164
Variasi dari modifikasi 1 dataset Tipe 1: 77.93116883116882

Standar deviasi dari modifikasi 2 dataset Tipe 1: 1.4713105347800273
Variasi dari modifikasi 2 dataset Tipe 1: 2.1647546897546897


### Langkah 3: Analisis

Dari beberapa perhitungan diatas kita peroleh hasil sebagai berikut:

Properti asli              : standar deviasi 4.41 dan variasi 19.48

Modifikasi 1 (perkalian 2) : standar deviasi 8.82 dan variasi 77.93 --> ($8.82 = 4.41\times2$)

Modifikasi 2 (pembagian 3) : standar deviasi 1.47 dan variasi 2.16 --> ($1.47 = 4.41/3$)

Dari data tersebut, bisa kita lihat (dan simpulkan) bahwa standar deviasi dari perkalian (atau pembagian) konstanta kepada setiap data dalam dataset sama dengan standar deviasi dari dataset yang dikali (atau dibagi) konstanta tersebut. Sedangkan untuk variasi tinggal menguadratkan standar deviasinya. Hal ini berarti perubahan data akan memengaruhi distribusi data sebesar konstanta yang digunakan. Jika konstanta yang digunakan lebih besar maka perbedaan distribusi data akan lebih besar.

## Properti 3: Sensitif terhadap Outlier

### Langkah 1: Memodifikasi dataset

Misal kita lakukan modifikasi dimana terdapat mahasiswa baru super jenius yang mendapat nilai 1000 di tiap tipe ujian, kita tambahkan **1 baris data dengan nilai 1000** ke dalam dataset.

In [7]:
# Menambahkan satu baris data dengan nilai 1000 untuk setiap kolom tipe ujian
new_row = pd.DataFrame({col: [1000] for col in df.columns})
df = pd.concat([df, new_row], ignore_index=True)

# Menghapus baris terakhir yang ditambahkan
# Gunakan agar tidak ada baris baru ganda jika program dijalankan berulang kali
# df = df.iloc[:-1] 

# Menampilkan 5 baris terakhir dari dataset modifikasi outlier
print("\nDataset modifikasi outlier:")
print(df.tail())


Dataset modifikasi outlier:
    Tipe1  Tipe2  Tipe3  Tipe4
52     15     10      8      2
53      4     20     12      5
54     15      6      5      4
55     20      5      3      6
56   1000   1000   1000   1000


### Langkah 2: Menghitung nilai standar deviasi dan variasi dari data yang sudah dimodifikasi

In [8]:
# Menghitung standar deviasi dan variasi untuk modifikasiol di dataset "Tipe 1"
std_modifol = df['Tipe1'].std()
var_modifol = df['Tipe1'].var()
# Menampilkan hasil standar deviasi dan variasi
print("Standar deviasi dari modifikasi outlier dataset Tipe 1:", std_modifol)
print("Variasi dari modifikasi outlier dataset Tipe 1:", var_modifol)

Standar deviasi dari modifikasi outlier dataset Tipe 1: 131.48768012655276
Variasi dari modifikasi outlier dataset Tipe 1: 17289.01002506266


### Langkah 3: Analisis

Dari beberapa perhitungan diatas kita peroleh hasil sebagai berikut:

Properti asli      : standar deviasi 4.41 dan variasi 19.48

Modifikasi outlier : standar deviasi 131.48 dan variasi 17289.01

Dari data tersebut bisa kita lihat bahwa penambahan outlier memberikan perubahan yang besar pada standar deviasi dan variasi. Hal ini karena nilai outlier menghasilkan selisih yang sangat besar dengan rata rata dataset, sehingga membuat standar deviasi meningkat secara drastis. Sementara variasi juga meningkat karena merupakan kuadrat dari standar deviasi.

Sehingga bisa kita simpulkan bahwa standar deviasi dan variasi memiliki sifat yang sama dengan mean, yaitu sensitif terhadap outlier.


